cheddar/cheddar-server/src/main/java/com/clicktravel/cheddar/server/rest/CheddarApiListingResource.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2014 Click Travel Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.clicktravel.cheddar.server.rest;
+
+import java.util.Set;
+
+import javax.servlet.ServletConfig;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.*;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.swagger.config.FilterFactory;
+import io.swagger.config.Scanner;
+import io.swagger.config.ScannerFactory;
+import io.swagger.config.SwaggerConfig;
+import io.swagger.core.filter.SpecFilter;
+import io.swagger.core.filter.SwaggerSpecFilter;
+import io.swagger.jaxrs.Reader;
+import io.swagger.jaxrs.config.BeanConfig;
+import io.swagger.jaxrs.config.JaxrsScanner;
+import io.swagger.jaxrs.listing.SwaggerSerializers;
+import io.swagger.models.Swagger;
+
+/**
+ * Cheddar's own version of the class com.wordnik.swagger.jaxrs.listing.ApiListingResource that removes the dependacy on
+ * a servlet deployment which cheddar currently do not support. This version changes the fetching of the swagger object
+ * to the scanner (BeanConfig) object instead of the servlet context.
+ */
+
+@Path("/")
+public class CheddarApiListingResource {
+
+    Logger LOGGER = LoggerFactory.getLogger(CheddarApiListingResource.class);
+
+    static boolean initialized = false;
+
+    protected synchronized Swagger scan(final Application app, final ServletConfig sc) {
```

Comment:
What is the purpose of this method? Make the code self-documenting to some extent.